File: semgus-sketch/src/main/java/org/semgus/sketch/ir/Rule.java
package org.semgus.sketch.ir;


import org.semgus.sketch.base.Expr;

import java.util.List;
import java.util.Map;

public record Rule(
  NonTerminal nt,
  List<NonTerminal> childNTs,
  Map<String, Expr> bindings  // output variable names -> bindings
) {}

File: semgus-sketch/src/main/java/org/semgus/sketch/ir/Rule.kt
package org.semgus.sketch.ir

import org.semgus.sketch.syntax.Expr

/**
 * IR rule.
 */
internal data class Rule(
  val nt: NonTerminal,
  val childNTs: Sequence<NonTerminal>,
  val binds: Map<String, Expr>,
  val ranks: Map<String, Int>
)
